### PR TITLE
Add more functions to register hasSideEffects => true

### PR DESCRIPTION
**Pull request by samsonasik**
The following functions has side effects:

```
stream_filter_append
ob_get_clean
openssl_sign
```

### CODE REVIEW

**Comment by staabm** [comment]
could you add a NodeScopeResolverTest regression test?
(especially for `ob_get_clean` which is widely used I think)

**Comment by samsonasik** [comment]
I will try when I have a chance.

I think between phpstan and psalm can complete each other side effect function lists, if we can have a script to include and mix both, that will be great for both sides.

File: bin/functionMetadata_original.php [hunk 1]
```diff
@@ -114,6 +115,9 @@
 	'random_bytes' => ['hasSideEffects' => true],
 	'random_int' => ['hasSideEffects' => true],
 
+	'ob_get_clean' => ['hasSideEffects' => true],
```
Comment (staabm) [inline bin/functionMetadata_original.php:118]: @samsonasik you can use something like https://phpstan.org/r/9632df30-9ef5-4c50-b070-32972826e7d4 to test this change in a `NodeScopeResolverTest`

the dumped type should be `false|string`

Comment (samsonasik) [reply bin/functionMetadata_original.php:118]: `ob_start()` is required to make a `false|string` https://phpstan.org/r/1cff680a-416e-4ffb-adaf-db73a67ef759

Comment (staabm) [reply bin/functionMetadata_original.php:118]: ob_start() could be anywhere, its not relevant for type analysis. Its global state. 

https://phpstan.org/r/a07431d0-a49d-47ff-bd7e-e432fec503f7

**Comment by samsonasik** [comment]
I am closing for now as I don't have a chance to update test yet.

**Pull request closed without merging**